contracts/SimpleProviders/DealProvider/DealProvider.sol

```diff
@@ -32,13 +35,14 @@ contract DealProvider is DealProviderModifiers, BasicProvider {
     function split(
         uint256 oldPoolId,
         uint256 newPoolId,
-        uint256 splitAmount
+        uint256 ratio
     )
         public
         override
         onlyProvider
-        invalidSplitAmount(poolIdToAmount[oldPoolId], splitAmount)
     {
+        uint256 splitAmount = poolIdToAmount[oldPoolId].calcAmount(ratio);
+        require(poolIdToAmount[oldPoolId] >= splitAmount, "Split amount exceeds the available amount");
```

Comment:
if we check that the ratio can't be > 1e18 - how can we get here?

Reply:
If some other provider is using it.

Reply:
Miscalculation of the provider, which calls the split function.